src/Psalm/Codebase.php

```diff
@@ -1422,12 +1423,19 @@ public function getCompletionItemsForClassishThing(string $type_string, string $
                             null,
                             (string)$method_storage->visibility,
                             $method_storage->cased_name,
-                            $method_storage->cased_name . (count($method_storage->params) !== 0 ? '($0)' : '()'),
+                            null,
                             null,
                             null,
                             new Command('Trigger parameter hints', 'editor.action.triggerParameterHints')
                         );
-                        $completion_item->insertTextFormat = \LanguageServerProtocol\InsertTextFormat::SNIPPET;
+                        if (count($method_storage->params) === 0) {
```

Comment:
I'd change this condition to `if (count(...) === 0 || !$snippets_supported)`. Sometimes you want to select a completion and proceed typing without the `(`. In those cases the `(` from the final `else` block would get in the way.

Also, not including `(` would be consistent with every other YCM completer.

Reply:
Consistency with other completers seems good here, but I'm not quite sure what that would be. When placeholders are supported and there are arguments, appending `($0)` seems obvious. I think you are suggesting to merge the other two cases (i.e. no placeholder support or no arguments), but what would you append then? Always `()`? Or nothing? I would think that appending `()` for a function that has arguments would be annoying, since you'd have to move the cursor back after insertion? Or is there something else going on here?

Reply:
> but what would you append then? Always ()? Or nothing?

Nothing. Just the function name and no parenthesis.

Reply:
Ok, thanks. I implemented that now, so now:
 - When snippets are supported and there are arguments, append `($0)`
 - Otherwise, append nothing.

This also allowed simplifying the code again, to just set the function name unconditionally in the constructor and optionally append the `($0)` later.

Thanks for your suggestions!